manifests/bootstrap_rpm.pp

```diff
@@ -0,0 +1,66 @@
+# This class builds an RPM containing the bootstrap for a subscription-manager consumer
+# This file is placed in $katello_www_pub_dir.
+# @api private
+class foreman_proxy_content::bootstrap_rpm (
+  Stdlib::Fqdn $hostname = $facts['networking']['fqdn'],
+  String $deployment_url = '/rhsm',
+  Stdlib::Port $rhsm_port = 443,
+  Stdlib::Absolutepath $rhsm_ca_dir = '/etc/rhsm/ca',
+  String $candlepin_cert_rpm_alias_filename = 'katello-ca-consumer-latest.noarch.rpm',
+  Stdlib::Absolutepath $katello_www_pub_dir = '/var/www/html/pub',
+) {
+  include certs
+
+  $katello_server_ca_cert = $certs::katello_server_ca_cert
+  $server_ca_name = $certs::server_ca_name
+  $default_ca_name = $certs::default_ca_name
+  $ca_cert = $certs::ca_cert
+  $server_ca = $certs::server_ca
+
+  $katello_rhsm_setup_script = 'katello-rhsm-consumer'
+  $katello_rhsm_setup_script_location = "/usr/bin/${katello_rhsm_setup_script}"
+
+  $candlepin_consumer_name = "katello-ca-consumer-${hostname}"
+  $candlepin_consumer_summary = "Subscription-manager consumer certificate for Katello instance ${hostname}"
+  $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
+
+  include trusted_ca
+  trusted_ca::ca { 'katello_server-host-cert':
+    source  => $katello_server_ca_cert,
+    require => File[$katello_server_ca_cert],
+  }
+
+  include apache
+  file { $katello_www_pub_dir:
+    ensure => directory,
+    owner  => 'apache',
+    group  => 'apache',
+    mode   => '0755',
+  } ->
+  # Placing the CA in the pub dir for trusting by a user in their browser
+  file { "${katello_www_pub_dir}/${server_ca_name}.crt":
+    ensure  => file,
+    source  => $katello_server_ca_cert,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    require => File[$katello_server_ca_cert],
+  } ~>
+  foreman_proxy_content::rhsm_reconfigure_script { "${katello_www_pub_dir}/${katello_rhsm_setup_script}":
```

Comment:
Maybe? But I am trying not to over think redesigning this due to how much this is going to change when it gets re-factored away from katello-certs-tools under the hood and towards both a simpler puppet layer and using RPM directly. I know that's hard, but instead of big bang, I'm trying to find the baby steps I can take to allow easier review while maintaining functionality